client/fingerprint/network_test.go

```diff
@@ -301,7 +302,9 @@ func TestNetworkFingerPrint_default_device(t *testing.T) {
 	}
 }
 
-func TestNetworkFingerPrint_excludelo_down_interfaces(t *testing.T) {
+// This tests that we prefer skipping link local and down interfaces as well as
+// those without global unicast addresses
+func TestNetworkFingerPrint_preferential_interfaces(t *testing.T) {
```

Comment:
I don't see how any of these tests are exercising the new code path with the scoring introduced above. 
I realize its not possible to test in a portable way across machines, but I would try to refactor the code so the scoring/selection logic is testable